wgengine/userspace.go

```diff
@@ -1341,32 +1341,32 @@ func (e *userspaceEngine) Ping(ip netaddr.IP, cb func(*ipnstate.PingResult)) {
 	e.magicConn.Ping(ip, cb)
 }
 
-func (e *userspaceEngine) RegisterIPPortIdentity(ipport netaddr.IPPort, n *tailcfg.Node) {
+func (e *userspaceEngine) RegisterIPPortIdentity(ipport netaddr.IPPort, tsIP netaddr.IP) {
 	e.mu.Lock()
 	defer e.mu.Unlock()
-	if e.nodeByIPPort == nil {
-		e.nodeByIPPort = make(map[netaddr.IPPort]*tailcfg.Node)
+	if e.tsIPByIPPort == nil {
+		e.tsIPByIPPort = make(map[netaddr.IPPort]netaddr.IP)
 	}
-	e.nodeByIPPort[ipport] = n
+	e.tsIPByIPPort[ipport] = tsIP
 }
 
 func (e *userspaceEngine) UnregisterIPPortIdentity(ipport netaddr.IPPort) {
 	e.mu.Lock()
 	defer e.mu.Unlock()
-	if e.nodeByIPPort == nil {
+	if e.tsIPByIPPort == nil {
 		return
 	}
-	delete(e.nodeByIPPort, ipport)
+	delete(e.tsIPByIPPort, ipport)
 }
 
-func (e *userspaceEngine) WhoIsIPPort(ipport netaddr.IPPort) (n *tailcfg.Node, ok bool) {
+func (e *userspaceEngine) WhoIsIPPort(ipport netaddr.IPPort) (tsIP netaddr.IP, ok bool) {
 	e.mu.Lock()
 	defer e.mu.Unlock()
-	if e.nodeByIPPort == nil {
-		return nil, false
+	if e.tsIPByIPPort == nil {
```

Comment:
you can delete these three lines.  it's valid to look things up in a nil map... it just returns zero value and false.